plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -65,7 +65,8 @@
     private final Optional<DataSize> maxScannedFileSize;
 
     @JsonCreator
-    public IcebergTableHandle(
+    @Deprecated // For JSON deserialization only
```

Comment:
Don't misuse `@Deprecated` like this. Name the method `fromJsonForDeserializationOnly`.